grpc-toolfile.spec

```diff
@@ -0,0 +1,60 @@
+### RPM external grpc-toolfile 1.0
+Requires: grpc
+
+%prep
+
+%build
+
+%install
+mkdir -p %i/etc/scram.d
+cat << \EOF_TOOLFILE >%i/etc/scram.d/grpc.xml
+<tool name="grpc" version="@TOOL_VERSION@">
+  <info url="https://github.com/grpc/grpc"/>
+  <lib name="grpc"/>
+  <lib name="grpc++"/>
+  <lib name="grpc++_reflection"/>
+  <lib name="absl_algorithm"/>
+  <lib name="absl_atomic_hook"/>
+  <lib name="absl_bad_optional_access"/>
+  <lib name="absl_base"/>
+  <lib name="absl_base_internal"/>
+  <lib name="absl_bits"/>
+  <lib name="absl_civil_time"/>
+  <lib name="absl_compressed_tuple"/>
+  <lib name="absl_config"/>
+  <lib name="absl_core_headers"/>
+  <lib name="absl_dynamic_annotations"/>
+  <lib name="absl_endian"/>
+  <lib name="absl_errno_saver"/>
+  <lib name="absl_inlined_vector"/>
+  <lib name="absl_inlined_vector_internal"/>
+  <lib name="absl_int128"/>
+  <lib name="absl_log_severity"/>
+  <lib name="absl_memory"/>
+  <lib name="absl_optional"/>
+  <lib name="absl_raw_logging_internal"/>
+  <lib name="absl_span"/>
+  <lib name="absl_spinlock_wait"/>
+  <lib name="absl_str_format"/>
+  <lib name="absl_str_format_internal"/>
+  <lib name="absl_strings"/>
+  <lib name="absl_strings_internal"/>
+  <lib name="absl_throw_delegate"/>
+  <lib name="absl_time"/>
+  <lib name="absl_time_zone"/>
+  <lib name="absl_type_traits"/>
+  <lib name="absl_utility"/>
+  <lib name="absl_meta"/>
+  <lib name="cares"/>
+  <lib name="address_sorting"/>
```

Comment:
@mrodozov , we do not need these libs in here. we should only have grpc libs defined here (they are linked to these linabsl_* libs and dynamic loader will take care of loading abls when needed).

@kpedro88 , there are few libgrps* libs. how are you planing to use them [a]. Should they all be in one tool file? If you know that you only need `grpc++` or `grpc` then we can start with those.  

[a]
libgrpc++_alts.so  libgrpc++_error_details.so  libgrpcpp_channelz.so    libgrpc.so    libgrpc_unsecure.so
libgrpc_cronet.so  libgrpc_plugin_support.so   libgrpc++_reflection.so  libgrpc++.so  libgrpc++_unsecure.so

Reply:
@smuzaffar we need all of these libraries (from abseil and grpc) for downstream products such as Triton.

We can try to make a separate spec for abseil - I mentioned this earlier in the thread, but did not get a response.

Reply:
having these libs in grpc distribution is fine but I think they should not be part of grpc tool file. 
@mrodozov , please create a separate tool file.

Reply:
@smuzaffar that is much easier, thanks for the suggestion
is it fine to keep them in the same install area (GRPC_BASE)?

Reply:
I separated only absl into different toolfile. I left the cares and address_sorting in . Or should I separate them into ? And can we keep the install area as suggested ?

Reply:
@kpedro88 , yes it is find to keep them in the grpc/lib
@mrodozov , all these absl, c-ares and address_sorting are externals which grpc build internal. I will suggest to move all of them out  in separate toolfiles e.g. abseil-cpp.xml, c-ares.xml, address_sorting.xml

Reply:
checking the build area again, I am not sure if we toolfile for abseil or address_sorting? The headers files of these tools are not exported (copied to grpc/include. So @kpedro88 I do not understand how you are going to use absl libs? Looks like these are internall used by grpc (and grpc libs are linked to them) , so there is no need to have them in toolfile.

Reply:
When I build Triton, it fails unless the absl libs are kept.

Reply:
The underlying problem is that `build/slc7_amd64_gcc820/external/grpc/1.28.1-cms/lib/cmake/grpc/gRPCTargets.cmake` includes the absl libs as targets (as well as cares and address_sorting)

Reply:
this is fine, we will keep the libs in distribution (we have to as grpc libs are linked to them). I am only suggesting to drop them from scram toolfile.

Reply:
Okay, I'll run the Triton installation that way and make sure it works.

Reply:
@smuzaffar @mrodozov the Triton installation works, so we can simplify by removing third party libs from the grpc toolfile.